components/client/src/main/java/com/hotels/styx/client/connectionpool/ConnectionExpirationTracker.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (C) 2013-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.client.connectionpool;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.base.Ticker;
+import com.hotels.styx.api.HttpRequest;
+import com.hotels.styx.api.HttpResponse;
+import com.hotels.styx.api.client.Connection;
+import com.hotels.styx.api.client.Origin;
+import rx.Observable;
+
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Provides wrapper for connection, that stores a connection expiration time. Also provides a method for verification of
+ * a time pasted.
+ */
+class ConnectionExpirationTracker implements ConnectionUsageTracker {
+    private final long connectionExpirationSeconds;
+    private final Supplier<Ticker> tickerSupplier;
+
+    ConnectionExpirationTracker(long connectionExpirationSeconds, Supplier<Ticker> tickerSupplier) {
+        this.connectionExpirationSeconds = connectionExpirationSeconds;
+        this.tickerSupplier = tickerSupplier;
+    }
+
+    @Override
+    public ExpirableConnection decorate(Connection connection) {
+        requireNonNull(connection);
+        return new TrackedConnectionAdapter(connection, connectionExpirationSeconds, tickerSupplier);
+    }
+
+    @Override
+    public boolean shouldTerminate(Connection connection) {
+        return connection instanceof ExpirableConnection && ((ExpirableConnection) connection).isExpired();
+    }
```

Comment:
We need to remove this method as discussed in my other comments. Also, see below...